index.coffee

```diff
@@ -20,21 +20,24 @@ tryExtractDigestHash = (evt) ->
 		matchPull = evt.status.match(/^Digest:\s([a-zA-Z0-9]+:[a-f0-9]+)$/)
 		return matchPull[1] if matchPull?
 
-awaitRegistryStream = (stream, onProgress) ->
+awaitRegistryStream = (stream, onProgress, ignoreErrorEvents) ->
 	contentHash = null
 	new Promise (resolve, reject) ->
 
 		jsonStream = JSONStream.parse()
 
 		jsonStream.on 'data', (evt) ->
 			if typeof evt isnt 'object'
-				evt = {}
-			else
+				return
```

Comment:
The previous behaviour was to call the `onProgress` handler with an empty object (`{}`). I don't see the point, hence simply skipping it (`return`). Either way, the non-object value is ignored. Also, I did not come across any example where `evt isnt 'object'`, so it may not really matter.

Reply:
I personally don’t remember at all why this conditional was put in place, but if it’s there it is because it was needed. The fact (?) that newer docker daemons may not ever send anything other than object doesn’t mean previous versions don’t as well. I would personally keep that check in because docker-progress is used with a very wide range of docker versions.

Reply:
Ah, wait I misread the diff, you’re not removing the check. I still can’t think of a reason why it’s there, given it’s not even forwaded as-is. I think removing it is fine then :)